rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -188,6 +189,17 @@ rmw_create_publisher(
     return nullptr;
   }
 
+  switch (qos_policy) {
+    case RMW_QOS_POLICY_BEST_EFFORT:
+      datawriter_qos.reliability.kind = DDS::BEST_EFFORT_RELIABILITY_QOS;
+      break;
+    case RMW_QOS_POLICY_RELIABLE:
+      datawriter_qos.reliability.kind = DDS::RELIABLE_RELIABILITY_QOS;
+      break;
+    default:
+      throw std::runtime_error("Unknown QoS policy");
```

Comment:
It would be good if the error string would include the actual numeric value of `qos_policy`.

Same below.